### PR TITLE
Dataset support

### PR DESCRIPTION
**Pull request by aviadlazar**
Stable datasets. The code will concat datasets of different videos for ease of use, while still allowing to iterate on each video separately.

### CODE REVIEW

File: SoiUtils/datasets/datasets.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            item = image_detection_sample
+        
+        return item.image, [det.__dict__ for det in item.detections]
+
+    def __len__(self):
+        return len(self.image_ids)
+    
+
+class ImageDetectionDatasetCollection(Dataset):
+    def __init__(self, collection_root_dir: str, annotation_file_name: str, **kwargs) -> None:
```
Comment (sharon200102) [inline SoiUtils/datasets/datasets.py]: Why would you assume that the annotation file name is identical in each of the datasets, I think annotation_file_name should be renamed and accept a list of names.

Comment (sharon200102) [reply SoiUtils/datasets/datasets.py]: Great work!

Comment (aviadlazar) [reply SoiUtils/datasets/datasets.py]: Solved in the next commit

File: SoiUtils/datasets/datasets.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    def __len__(self):
+        return len(self.collection)
+
+    def get_sub_dataset(self, index):
+        return self.collection.datasets[index]
+    
+    def num_subsets(self):
+        return len(self.collection_items_root_dirs)
+
+train_dataset = ImageDetectionDatasetCollection(Path('test_dataset/train'), ["annotations.json"]*4)
```
Comment (sharon200102) [inline SoiUtils/datasets/datasets.py]: Please remove these lines of code, I think they were meant for testing.

File: SoiUtils/test.py
Comment (sharon200102) [inline SoiUtils/test.py:1]: This test file is not necessary any more, I think it should be removed.

**Pull request merged**